### DISCUSSION TITLE
"Open questions" feature?

### DISCUSSION THREAD

**Discussion by hew-wolff** [pi-base/data#487] [Ideas]
Can we use the data to suggest interesting open questions? The questions would be potential theorems consistent with the data. Not sure what "interesting" would mean exactly, but probably it would prefer shorter theorems.

**Comment by StevenClontz** [comment]
agree 100%. Support for searching for unknown traits is in the preview at https://topology.pages.dev/spaces?q=%24T_2%24%2B%3FMetrizable but a more direct "roll dice for your undergraduate thesis topic" button would be swell.

**Comment by jamesdabbs** [comment]
Yeah, definitely agree that we should be able to do something valuable here. Do y'all have a feel for whether you'd prefer?:

* Literally random not-disprovable implication
* Pick a specific property you're interested in and get a theorem relating to that property
* ... something else ...

Or is this a "we need to build something and feel it out" situation?

**Reply by StevenClontz** [reply]
Right now searches like https://topology.pages.dev/spaces?q=Retracts+are+closed%2B%7EUS need a better call to action (e.g. if `P+Q+R` returns no results, we provide a link to a GitHub issue template that suggests adding a space with those properties or adding a theorem of the form `P+Q=>~R`).

Then our randomizer could reveal such a search.

**Reply by hew-wolff** [reply]
I think ideally there would be a search with various parameters. Completely random is a good start, but it would also be nice to (as you say) filter it to a property, or sort by some "interesting" criteria. I guess the spaces and properties are already sorted in a helpful way.

I haven't honed my skills for the kind of manual exploration suggested by @prabau but that seems like an excellent complement to such a search.

**Comment by prabau** [comment]
This is all a nice idea in theory.  But in my experience, one can do the same thing by manual exploration of the database. If a student (or anyone) is interested in a certain property, they can very easily bring up a list of examples or counterexamples, then cross reference this with other related properties or their negations, etc, and find if there are examples or counterexamples or nothing available for any number of combinations of properties and their negations.  And then investigate why there is no such example, maybe because of a missing theorem that could be added, or find an illustrating space that could be added.  Or a theorem could be generalized to cover more cases.  Etc, etc.  That's how I usually proceed and pi-base is a wonderful tool for this kind of exploration.  There are tons and tons of things that come up just by this kind of manual exploration.

I would even argue that such a manual exploration provides a more in depth understanding of the concepts involved, instead of having an algorithm spitting out some random missing thing.  Especially as the student can focus on groups of related properties they are interested in, and couple that with literature search and research.

**Reply by StevenClontz** [reply]
I think the randomizer is more about breaking the ice, and as it was requested by at least a couple participants in last week's workshop involved in similar projects I'd like to see it happen. Also, I certainly could see myself using it during demos when I really do just need a random example to work with.

**Comment by prabau** [comment]
On the other hand I think that adding more convenient ways to inspect the data would be very valuable.

In particular a comparator for spaces where the user would pick two spaces and get a table showing all their properties side by side, with indication of asserted/deduced/missing.  An undergraduate could look at that and investigate and try to fill missing slots for two closely related spaces.  Just an idea.

**Reply by StevenClontz** [reply]
Definitely agree that having a view like `/spaces/[id]/compare/[otherId]` would be handy.
